dev/devicelab/lib/tasks/microbenchmarks.dart

```diff
@@ -55,6 +58,33 @@ TaskFunction createMicrobenchmarkTask() {
     allResults.addAll(await _runMicrobench('lib/gestures/velocity_tracker_bench.dart'));
     allResults.addAll(await _runMicrobench('lib/stocks/animation_bench.dart'));
 
+    // Run micro-benchmarks once again in --preview-dart-2 mode.
+    // Append "_dart2" suffix to the result keys to distinguish them from
+    // the original results.
+
+    void addDart2Results(Map<String, double> benchmarkResults) {
+      benchmarkResults.forEach((String key, double result) {
+        allResults[key + '_dart2'] = result;
```

Comment:
`dart2` suffix is different from `preview_dart_2` we are using for hot mode tests, which makes it slightly more difficult to see all dart 2.0 benchmarks.  Can you consider changing it?

Also, for hot mode benchmarks, `preview_dart_2` goes into the name of the benchmark, rather into the name of the result. Not sure if these microbenchmarks can be made consistent with hot mode ones in that regard as well.

Reply:
@aam, I did exactly what you are suggesting in the [first version](https://github.com/flutter/flutter/pull/13225/commits/d3c6cd83bcb0f1e4ad1b831b0873ae67f4b2a69f) of the change.

@yjbanov asked me to change it in the [review](https://github.com/flutter/flutter/pull/13225#discussion_r153613625).

I think it's OK to change the suffix in the result to 'preview_dart_2', but creating a benchmark with different name wastes horizontal space on the dashboard and should be avoided.

Also, I see other benchmarks such as 'flutter_gallery_win__compile' which have both default mode and '_preview_dart_2' results.

Reply:
Ah, thanks @alexmarkov , didn't see @yjbanov review comments regarding that. 

>I think it's OK to change the suffix in the result to 'preview_dart_2', but creating a benchmark with different name wastes horizontal space on the dashboard and should be avoided.

Are you talking https://flutter-dashboard.appspot.com/benchmarks.html dashboard? Not sure how it works but it looks max of benchmark name and result name that determines card width. I would argue that you can reduce font size if you want to fit more benchmarks on the screen. Consistency in benchmarks naming(especially considering filtering that is done using name text) seems more important.

> Also, I see other benchmarks such as 'flutter_gallery_win__compile' which have both default mode and '_preview_dart_2' results.

Yes, I see it now. I think it's unfortunate because the name of the result(`aot_snapshot_size`, for example) is the same, it's the benchmark(`win_compile with preview_dart_2`) that is different. Not sure whether there was some particular reason for attributing result name, rather than the benchmark, @Hixie ?

Reply:
The benchmark numbers for dart2 for compiling are just obtained at the same time as we do all the other compiling benchmarks (debug, release, aot size, binary size, etc). It's all just one big benchmark, run on three platforms, for each of three apps.

We should probably just move the hot mode dart2 benchmarks into the one hot mode test in the same way (not that it really matters either way).

Reply:
Look at it differently - running flutter in preview-dart-2 mode can behave(pass/fail) differently from base mode. Grouping them together will be misleading.
Results of the run(if the run was successful) are named the same(aot size, compile time, run time), but have to grouped in such a fashion that it is clear what run({base|preview-dart-2} x {ios|android} x {linux|win|mac}) they belong to.

Reply:
Let's keep the "dart2" benchmark IDs. I'll update the UI labels (this is done in the database, not in code)

Reply:
> Look at it differently - running flutter in preview-dart-2 mode can behave(pass/fail) differently from base mode. Grouping them together will be misleading.

I don't really see why. Either the build is green or the build is red; if the build is red we should back out what made it red. Doesn't really matter what exactly caused it to go red.

> Results of the run(if the run was successful) are named the same(aot size, compile time, run time), but have to grouped in such a fashion that it is clear what run({base|preview-dart-2} x {ios|android} x {linux|win|mac}) they belong to.

They are (insofar as anything can be grouped across multiple dimensions when it's a single list). The names are very consistent now for all these.

Reply:
> I don't really see why. Either the build is green or the build is red; if the build is red we should back out what made it red. Doesn't really matter what exactly caused it to go red.

I think preview-dart-2 builds have more leeway regarding being red. Just recently(during last dart roll) we have marked preview-dart-2 hot reload flaky because it was broken(even if just for one evening) by dart roll, but we didn't want to keep rope red because of that.

Reply:
If the code is in, it should work. We don't give more leeway for other not-yet-final parts of the code, why would dart2 be special?

Reply:
> Let's keep the "dart2" benchmark IDs. I'll update the UI labels (this is done in the database, not in code)
 
FWIW, I find it really confusing when the labels don't match the code (because I can't search for the label in the code).

Reply:
What if we add labels to `manifest.yaml` and use stable numeric IDs for all benchmarks? Then there will be _one_ name both in code and on the UI. WDYT?

Reply:
That could work.